Added plan summaries and updated overview figure

mvonsieb opened the pull request:
According to discussion by mail and in RTTD Terminlogy Working Group 2022-01-11
- Updated profile names (except for prescription I did not edit in the ‘cumulative’)
- Updated the profile overview figure
- Removed number of Sessions from Course Prescription
- Added energy and device type under modalityAndTechnique in RequestCommon, Course Summary and Treated Phase.
- Changed order of fractionDosePlanned in radiotherapy-dose-planned-to-volume so this new element it at the end and it looks more consistent with the delivered dose extension.
 
Additionally
 
- Changed names of extension elements in profiles consistent with mCODE, e.g.
- Modality-and-technique -> modalityAndTechnique
- Removed some outdated comments

Open:
  - Extend the examples to contain instances of the new profiles
   - I did not rename files yet so it’s easier to review. Will do that in a separate round after merging.
   - SAK:   value set for treatment device type needs to be created

mvonsieb committed:
added plan summaries and updated overview figure

mvonsieb committed:
USCoreServiceRequest only in laster US Core

saulakravitz reviewed (commented):
Per our discussion on 22/12:

* annotate overview figure to flag which data elements are extensions, and which extensions are defined by reference to mCODE
 * Profile names -- I think we settled on:
   * Planned Course --> Course Summary
   * Planned Phase --> Treated Phase
   * Plan --> Treated Plan
* [Dose Delivered to Volume](https://build.fhir.org/ig/HL7/fhir-mCODE-ig//StructureDefinition-mcode-radiotherapy-dose-delivered-to-volume.html)
  * Curious: Why is fractions delivered appropriate in mCODE, but not appropriate (0..0) in Treatment Phase (replaced by a simple extension)?
  * I think it is confusing to have a 2 field complex extension with one field 0..0.... better to add another simple codexrt extension for dose to volume that has only what you need.
* Make the order of sub-extensions in [DosePrescribedToVolume](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-dose-prescribed-to-volume.html) the same as the corresponding mCODE extension (the fractions and dose are in different order, and it requires unnecessary brain cycles to compare the two.).
* [RadiotheraphyPhasePrescription](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotheraphy-phase-prescription.html) and [RadotherapyPlannedPhase](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-planned-phase.html)
  * should energy and device both be MS?
  * should energy either be 0..1, or should an energy be associated to each modality?
  * Should the device be directly associated to the modality/technique?
 * [PlannedCourseSummary](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-planned-course-summary.html) and [RadotherapyPlannedPhase](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-planned-phase.html)
   * Should bodySite be bound to the TreatmentLocation VS?
 *

mvonsieb committed:
catch up with master

mvonsieb committed:
Merge remote-tracking branch 'origin/master' into add-plan-summaries

mvonsieb committed:
updates according to WG discussion 2022-01-11

mvonsieb commented:
> Per our discussion on 22/12:
> 
> * annotate overview figure to flag which data elements are extensions, and which extensions are defined by reference to mCODE
> * Profile names -- I think we settled on:
>   
>   * Planned Course --> Course Summary
>   * Planned Phase --> Treated Phase
>   * Plan --> Treated Plan
> * [Dose Delivered to Volume](https://build.fhir.org/ig/HL7/fhir-mCODE-ig//StructureDefinition-mcode-radiotherapy-dose-delivered-to-volume.html)
>   
>   * Curious: Why is fractions delivered appropriate in mCODE, but not appropriate (0..0) in Treatment Phase (replaced by a simple extension)?
>   * I think it is confusing to have a 2 field complex extension with one field 0..0.... better to add another simple codexrt extension for dose to volume that has only what you need.
> * Make the order of sub-extensions in [DosePrescribedToVolume](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-dose-prescribed-to-volume.html) the same as the corresponding mCODE extension (the fractions and dose are in different order, and it requires unnecessary brain cycles to compare the two.).
> * [RadiotheraphyPhasePrescription](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotheraphy-phase-prescription.html) and [RadotherapyPlannedPhase](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-planned-phase.html)
>   
>   * should energy and device both be MS?
>   * should energy either be 0..1, or should an energy be associated to each modality?
>   * Should the device be directly associated to the modality/technique?
> * [PlannedCourseSummary](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-planned-course-summary.html) and [RadotherapyPlannedPhase](http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries/StructureDefinition-codexrt-radiotherapy-planned-phase.html)
>   
>   * Should bodySite be bound to the TreatmentLocation VS?

According to discussion by mail and in RTTD Terminlogy Working Group 2022-01-11

- Updated profile names (except for prescription I did not edit in the ‘cumulative’)
- Removed number of Sessions from Course Prescription
- Added energy and device type under modalityAndTechnique in RequestCommon, Course Summary and Treated Phase.
- Changed order of fractionDosePlanned in radiotherapy-dose-planned-to-volume so this new element it at the end and it looks more consistent with the delivered dose extension.

Additionally

- Changed names of extension elements in profiles consistent with mCODE, e.g.
- Modality-and-technique -> modalityAndTechnique
- Removed some outdated comments

saulakravitz commented on input/fsh/SD_Extensions.fsh:
```diff
@@ -59,9 +63,38 @@ Description: "Dose parameters prescribed for one radiotherapy volume."
+    fractionDosePlanned 0..1 MS
+* extension[volume]
+  * value[x] only Reference(RadiotherapyVolume)
+  * ^short = "Volume to which Dose is Planned"
```
Perhaps this should be:
"Volume targeted by planned dose"?
Or perhaps "Volume for which dose is planned".

mvonsieb replied:
Changed to "Volume for which dose is planned". But left "Volume to which dose is prescribed" in the prescription, right?

saulakravitz replied:
I would stick to the same pattern:   Volume targeted by the prescribed dose.

mvonsieb replied:
Ok, fixed to " Volume targeted by the prescribed/planned dose."

saulakravitz commented on input/fsh/SD_Extensions.fsh:
```diff
@@ -59,9 +63,38 @@ Description: "Dose parameters prescribed for one radiotherapy volume."
+* extension[volume]
+  * value[x] only Reference(RadiotherapyVolume)
+  * ^short = "Volume to which Dose is Planned"
+  * ^definition = "A BodyStructure resource representing the body structure treated, for example, Chest Wall Lymph Nodes."
```
treated --> to be treated, or that is the target of treatment

mvonsieb replied:
> treated --> to be treated, or that is the target of treatment
Fixed to "to be treated".

saulakravitz commented on input/fsh/SD_Extensions.fsh:
```diff
@@ -59,9 +63,38 @@ Description: "Dose parameters prescribed for one radiotherapy volume."
+* extension[fractionsPlanned]
+  * value[x] only positiveInt
+  * ^short = "Number of Planned Fractions"
+  * ^definition = "The number of Fraction in which dose to this volume is planned. See also extension RadiotherapyFractionsPlanned which is used instead if fractions are the same for all volumes, i.e. in Planned Phases or Plans."
```
"The number of Fraction in which dose to this volume is planned"=>
  "The planned number of fractions to deliver the dose"?

mvonsieb replied:
> "The number of Fraction in which dose to this volume is planned"=> "The planned number of fractions to deliver the dose"?
Changed to "The planned number of Fractions to deliver the dose", and similar for "prescribed number of Fractions" in the respective extensions.

saulakravitz commented on input/fsh/SD_Extensions.fsh:
```diff
@@ -59,9 +63,38 @@ Description: "Dose parameters prescribed for one radiotherapy volume."
+Title: "Radiotherapy Energy"
 Description: "The radiation energy, preferably specified as decimal in MV. Can alternatively be represented as string or CodeableConcept."
 * . ^short = "Radiotherapy Energy"
 * value[x] ^short = "The radiation energy as decimal in MV. Can alternatively be represented as string or CodeableConcept."
```
Why do you want to leave the value type open?   how would it be represented as a codeable concept?
Shouldn't it be limited to a Quantity?   Like we do for dose?  With MV as the required units?

mvonsieb replied:
That was biased by the fact that in our system the Energy is an enumeration of selectable energies. We have to discuss in the terminology call if we allow only for discrete coded values for a ValueSet, so one cannot send 9.999999MV or 10.004MV, or define a precision, or don't care too much. Another use for codes would be to have the same extension for energy or isotope, but I never changed the description and in the meantime I think it should be different extensions. Let's also discuss that in the call.

mvonsieb replied:
Anyway, fixed to only Quantity for now to be inline with the description and scope that is only energy, not isotope so far.

saulakravitz commented on input/fsh/SD_RadiotherapyTreatmentPhase.fsh:
```diff
@@ -12,21 +12,24 @@ Please note that while these codes may imply conformance to the profiles that us
-* #codexrt-teleradiotherapy-treatment-phase "Teleradiotherapy Treatment Phase Resource" "Identifies a Procedure resource that describes delivery of teleradiotherapy (external beam radiation) and conforms to the TeleradiotherapyTreatmentPhase profile." // SNOMED Code Requested USCRS-33293
-* #codexrt-brachytherapy-treatment-phase "Brachytherapy Treatment Phase Resource" "Identifies a Procedure resource that describes delivery of brachytherapy (internal or surface radiation) and conforms to the BrachytherapyTreatmentPhase profile." // SNOMED Code Requested USCRS-33294
+* #codexrt-teleradiotherapy-treatment-phase "Teleradiotherapy Treatment Phase Resource" "Identifies a Procedure resource that describes delivery of teleradiotherapy (external beam radiation) and conforms to the TeleradiotherapyTreatedPhase profile." // SNOMED Code Requested USCRS-33293
+* #codexrt-brachytherapy-treatment-phase "Brachytherapy Treatment Phase Resource" "Identifies a Procedure resource that describes delivery of brachytherapy (internal or surface radiation) and conforms to the BrachytherapyTreatedPhase profile." // SNOMED Code Requested USCRS-33294
```
USCRS-33293 and 33294 have been rejected/withdrawn.
The only code currently requested is RadiotherapyTreatmentPhase USCRS-33527 (parent of brachy and teleradio treatment phases, should we request them).

mvonsieb replied:
Removed the complete outdated comment section. The USCRS-33527 is properly contained as requested code in DEF_CodeSystems.fsh.

saulakravitz committed:
alternate extension

saulakravitz commented on input/fsh/DEF_Rulesets.fsh:
```diff
@@ -1,13 +1,15 @@
 * extension contains
-    $mCODERadiotherapyModalityAndTechnique named modality-and-technique 0..* MS and
-    RadiotherapyDosePrescribedToVolume named radiotherapy-dose-prescribed-to-volume 0..* MS
+    $mCODERadiotherapyModalityAndTechnique named modalityAndTechnique 0..* MS
```
I think it may be better to define a new Codex-RT extension that is a profile of the mCODE modality and technique extension.   See http://build.fhir.org/ig/HL7/codex-radiation-therapy/branches/add-plan-summaries-sak/StructureDefinition-codexrt-radiotheraphy-phase-prescription.html

I think it is better because:
   - the defining URL of the new extension makes it clear that it is different than the mCODE extension
   - it preserves all of the invariants
   - there is a clickable entity in the IG that shows the content of the extension....you still have to click through it to see the content of the mCODE extension, but I think it looks cleaner.
     - I've asked Mark for his opinion.
     
My option works very *poorly* for CourseSummary where modality and technique is already baked in (which is probably why you went the path you did)....sigh....

mvonsieb replied:
The invariants would still be preserved if we'd use the original mCODE extension. I see the advantange of having a dedicated url for the complex extension that has everything (modality, technique, device type, energy or isotope (to be defined) and maybe more in the future). But since we don't have good name for it I was not so keen on that. On the other hand just having new extensions device-type, energy, isotope and not care about the context in the first place would be very simple as we discussed already. We just happen to use them here in the modality-and-technique hierarchy but in principle they would be self standing extensions for any context. I understand we could have both, self standing extensions for all elements without restricted context AND use them in a complex extension derived from the mCODE extension . But the simplest to me is still just new single element extensions and the resource profile just put them in the right place for these profiles, i.e. in the modality and technique mCODE extension.

saulakravitz replied:
I'm pretty much convinced that status quo is fine....still waiting for Mark to weigh in.

saulakravitz committed:
require values in extension.  Fix broken link

saulakravitz commented on input/fsh/SD_Extensions.fsh:
```diff
@@ -21,18 +16,27 @@ Device instances are not specified here. Those are represented by resources of t
 This extension SHALL only be present if the treatment is structured as countable Fractions, for example in a Phase Prescription."
 * . ^short = "Number of Prescribed Fractions"
 * value[x] ^short = "Number of Prescribed Fractions"
 * value[x] only positiveInt
```
General comment for all values:   They should be required 1..1.    The extension isn't required, but if you include the extension the value should be required.   See my variant of your branch with -sak suffix.

mvonsieb replied:
Agreed. All 1..1.

saulakravitz commented on input/fsh/SD_RadiotherapyPhasePrescription.fsh:
```diff
@@ -2,23 +2,23 @@
 Profile: RadiotherapyPhasePrescription
```
reasonCode is MS but not bound to anything.
Should reason Code be bound to a disease code in SNOMED/ICD-10?

saulakravitz reviewed (approved):
See my comments.

mvonsieb committed:
addressing review comments

saulakravitz committed:
tweak images

mvonsieb committed:
More review findings. Binding for reasonCode.

mvonsieb committed:
Update DEF_Rulesets.fsh

mvonsieb committed:
Update SD_Extensions.fsh

mvonsieb committed:
Merge branch 'add-plan-summaries' into add-plan-summaries-sak

mvonsieb committed:
Merge pull request #13 from HL7/add-plan-summaries-sak

Add plan summaries sak, merged together in call.
